### PR TITLE
Make ZHA config panel device oriented

### PR DESCRIPTION
**Pull request by dmulcahey**
This cleans up the ZHA config panel a bit and makes it device oriented. It removes entities from all ZHA operations and does everything on a device level. 

**DO NOT MERGE WITHOUT:** https://github.com/home-assistant/home-assistant/pull/20990

### CODE REVIEW

File: src/panels/config/zha/zha-node.ts
Comment (balloob) [inline src/panels/config/zha/zha-node.ts]: You want to do initial fetching in `firstUpdated()`

File: src/panels/config/zha/zha-device-card.ts
Comment (balloob) [inline src/panels/config/zha/zha-device-card.ts]: Can you add a message if there are 0 entities? Or is that impossible?

File: src/panels/config/zha/zha-device-card.ts
Comment (balloob) [inline src/panels/config/zha/zha-device-card.ts]: To make this more readable, you can put the `@click` on it's own line.

File: src/panels/config/zha/zha-device-card.ts
Comment (balloob) [inline src/panels/config/zha/zha-device-card.ts]: Not needed

File: src/panels/config/zha/zha-device-card.ts
Comment (balloob) [inline src/panels/config/zha/zha-device-card.ts]: We are super cool now, you can do this:

```suggestion
  @property() public hass?: HomeAssistant;
```

And then drop `static get properties()`

See #2711

File: src/panels/config/zha/zha-node.ts
Comment (balloob) [inline src/panels/config/zha/zha-node.ts]: Assigning twice _feels_ a bit weird, I would prefer:

```ts
    this._nodes = (await fetchDevices(this.hass!)).sort((a, b) => { … });
```

However, it's more like a nitpick, ignore if you want.

**Review by balloob — APPROVED** [review]
Can be merged when comments addressed.

File: src/panels/config/zha/zha-device-card.ts
Comment (dmulcahey) [reply src/panels/config/zha/zha-device-card.ts]: every device will have at least one

File: src/panels/config/zha/zha-node.ts
Comment (dmulcahey) [inline src/panels/config/zha/zha-node.ts]: @balloob I had to make a change here to clear selection and fetch fresh device list after remove is called. Just want to make sure the change is good before I merge this and the backend PR.

**Comment by dmulcahey** [comment]
All comments addressed and a new question added.

File: src/panels/config/zha/zha-node.ts
Comment (balloob) [reply src/panels/config/zha/zha-node.ts]: If a remove is successful, just remove the device on the client side instead of refreshing devices?

Comment (dmulcahey) [reply src/panels/config/zha/zha-node.ts]: I can do that. Thx

Comment (balloob) [reply src/panels/config/zha/zha-node.ts]: Going to merge this now so I can do a build, you can do it in another PR.

**Review by balloob — APPROVED** [review]

**Pull request merged**
